include/boost/regex/v4/mem_block_cache.hpp

```diff
@@ -19,17 +19,72 @@
 #define BOOST_REGEX_V4_MEM_BLOCK_CACHE_HPP
 
 #include <new>
-#ifdef BOOST_HAS_THREADS
-#include <boost/regex/pending/static_mutex.hpp>
-#endif
+#include <boost/atomic/atomic.hpp>
```

Comment:
This include should be inside the #ifdef BOOST_NO_CXX11_HDR_ATOMIC block to avoid unnecessary inclusion.